platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -372,8 +372,26 @@ export default class HangingProtocolService extends PubSubService {
    * for example, a prior view hanging protocol will NOT show the active study
    * specifically, but will show another study instead.
    */
-  public setActiveStudyUID(activeStudyUID: string): void {
+  public setActiveStudyUID(activeStudyUID: string) {
+    if (!activeStudyUID || activeStudyUID === this.activeStudy?.StudyInstanceUID) {
+      return;
+    }
     this.activeStudy = this.studies.find(it => it.StudyInstanceUID === activeStudyUID);
+    return this.activeStudy;
+  }
+
+  public hasStudyUID(studyUID: string): boolean {
+    return this.studies.some(it => it.StudyInstanceUID === studyUID);
+  }
+
+  public addStudy(study) {
+    if (!this.hasStudyUID(study.StudyInstanceUID)) {
+      this.studies.push(study);
+    }
+  }
```

Comment:
There are too many instances of `study` versus `studyUID` in this file. I'm unsure why we have both, and this is contributing to the confusion. Can we simplify it to work with `study`, as it seems sufficient for our needs?

```typescript
export interface StudyMetadata extends Record<string, unknown> {
  readonly StudyInstanceUID?: string;
  StudyDescription?: string;
}
```

so I propose `hasStudy(study)` of `hasStudyUID`